app/stores/AppStore.js

```diff
@@ -32,23 +34,30 @@ export default class AppStore extends Store {
 
   @computed get currentLocale(): string {
     if (!this.isCurrentLocaleSet) {
+      this.getProfileLocaleRequest.execute();
       return 'en-US';
     }
-    return this.profileLocaleRequest.result;
+    return this.getProfileLocaleRequest.result || this.setProfileLocaleRequest.result;
+  }
+
+  @computed get hasLoadedCurrentLocale(): bool {
+    return this.getProfileLocaleRequest.wasExecuted;
   }
 
   @computed get isCurrentLocaleSet(): bool {
-    return this.profileLocaleRequest.wasExecuted;
+    if (this.getProfileLocaleRequest.result == null) return false;
+    return (this.getProfileLocaleRequest.wasExecuted && this.getProfileLocaleRequest.result) ||
+           (this.setProfileLocaleRequest.wasExecuted && this.setProfileLocaleRequest.result);
```

Comment:
I see why you are doing it like that but i think it is cleaner if you invalidate and execute the `getProfileLocaleRequest` anytime the `setProfileLocaleRequest` was executed like this (in `_updateLocale` method):

`this.setProfileLocaleRequest.execute(locale).then(() => this.getProfileLocaleRequest.invalidate().execute())`

This way everything stays declarative and you don't need the complicated check (you can just ask for `this.getProfileLocaleRequest.result` to see if `isCurrentLocaleSet`)